ProcessHacker/tokprp.c

```diff
@@ -946,25 +946,38 @@ INT_PTR CALLBACK PhpTokenPageProc(
                         tokenPageContext->Context
                         )))
                     {
-                        if (NT_SUCCESS(status = PhGetTokenIntegrityLevel(
+                        if (NT_SUCCESS(status = PhGetTokenIntegrityLevelRID(
                             tokenHandle,
-                            &integrityLevel,
+                            &integrityLevelRID,
                             NULL
                             )))
                         {
+                            ULONG customLevelPosition = 0; // Processing atypical integrity levels
+
                             for (ULONG i = 0; i < menu->Items->Count; i++)
                             {
                                 PPH_EMENU_ITEM menuItem = menu->Items->Items[i];
 
-                                if (menuItem->Id == (ULONG)integrityLevel)
+                                if (menuItem->Id == (ULONG)integrityLevelRID)
                                 {
                                     menuItem->Flags |= PH_EMENU_CHECKED | PH_EMENU_RADIOCHECK;
+                                    customLevelPosition = 0; // The integrity level is a well-known one. No need to add a new menu item.
                                 }
-                                else if (menuItem->Id > (ULONG)integrityLevel)
+                                else if (menuItem->Id > (ULONG)integrityLevelRID)
                                 {
                                     menuItem->Flags |= PH_EMENU_DISABLED;
+                                    customLevelPosition = i + 1;
                                 }
                             }
+
+                            if (customLevelPosition)
+                            {
+                                PPH_EMENU_ITEM unknownIntegrityItem;
+                                
+                                unknownIntegrityItem = PhCreateEMenuItem(0, (ULONG)integrityLevelRID, L"Custom level", NULL, NULL);
```

Comment:
Maybe rename the menu "Intermediate level" since the level has an intermediate position?